src/Rules/ClassNameCheck.php

```diff
@@ -0,0 +1,31 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules;
+
+class ClassNameCheck
+{
+
+	public function __construct(
+		private ClassCaseSensitivityCheck $classCaseSensitivityCheck,
+		private ClassForbiddenNameCheck $classForbiddenNameCheck,
+	)
+	{
+	}
+
+	/**
+	 * @param ClassNameNodePair[] $pairs
+	 * @return RuleError[]
+	 */
+	public function checkClassNames(array $pairs, bool $checkClassCaseSensitivity = true): array
+	{
+		$errors = [];
+
+		if ($checkClassCaseSensitivity) {
+			$errors += $this->classCaseSensitivityCheck->checkClassNames($pairs);
```

Comment:
I don't really like doing `+` here. I'd rather iterate over the array and add the error one-by-one with `$errors[] = ...`.